lib/sinatra/base.rb

```diff
@@ -78,6 +78,8 @@ def params
       super
     rescue Rack::Utils::ParameterTypeError, Rack::Utils::InvalidParameterError => e
       raise BadRequest, "Invalid query parameters: #{Rack::Utils.escape_html(e.message)}"
+    rescue EOFError => e
+      raise BadRequest, "Invalid multipart/form-data: #{Rack::Utils.escape_html(e.message)}"
```

Comment:
It's not clear to me why you escape the HTML here, rather than on the output formatting side.

Reply:
I can only provide background info, not speak to if this is the best solution. The introduction of handling of `EOFError` probably just followed what was already there: the handling of invalid query parameters – escaping those was introduced in https://github.com/sinatra/sinatra/pull/1432 to address an reported XSS https://github.com/sinatra/sinatra/issues/1428 (the initial handling of invalid query parameters was added in https://github.com/sinatra/sinatra/pull/1070)

Feel free to make a PR if you have a more appropriate solution.

Reply:
I think it makes sense to escape here as otherwise every application outputting the `BadRequest` message would need to do it.

Reply:
You should escape any error messages regardless because you don't control every error message generation, right? It should be escaped when generating HTML from the error message, not when creating an exception. Otherwise, how do you  print this to the command-line, log file, etc. Maybe I'm misunderstanding something.

Reply:
Sure, what you say makes sense. Feel free to contribute and maybe others will weigh in as well.

Reply:
Thanks for mentioning. To be honest, I just copied lines above and changed error type and message. I didn't think about whether this usage of escape_html was correct or not.